controller.py

```diff
@@ -129,11 +141,29 @@ def update_url(time=last_update_time, page=None):
 
         return feed_response(update_feed)
 
-    def remove_items(self):
-        catalog = self.authenticated_catalog_from_request()
-        if isinstance(catalog, ProblemDetail):
-            return catalog
+    def register_client(self):
+        """Creates authentication details for a server"""
+        url = request.args.get('client_url')
+        if not url:
+            return INVALID_INPUT.detailed("No 'client_url' provided")
+
+        try:
+            url = urllib.unquote(url)
+            server, plaintext_secret = ClientServer.register(self._db, url)
+            self._db.commit()
+
+            body = json.dumps(dict(key=server.key, secret=plaintext_secret))
+            return make_response(body, 200, {"Content-Type": "application/json"})
+        except ValueError, e:
+            return INVALID_INPUT.detailed(e.message)
+
+    def remove_items(self, collection_details):
+        """Removes identifiers from a collection's catalog"""
+        server = authenticated_server_from_request(self._db)
+        if isinstance(server, ProblemDetail):
+            return server
 
+        collection = Collection.from_metadata_identifier(self._db, collection_details)[0]
```

Comment:
I use the `collection, ignore = Collection...` syntax in this situation to avoid giving the impression that `from_metadata_identifier` might return a large number of `Collection` objects and we're just picking the first one.